api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/cognito/AmazonCognitoAuthenticationProperties.java

```diff
@@ -56,4 +60,30 @@
      */
     @RequiredProperty
     private String userPoolId;
+
+    /**
+     * Describes the map of attributes that are to be fetched from the Cognito user pool (map keys)
+     * and then transformed/renamed using map values before they are put into a principal.
+     * An example might be to fetch {@code custom:custom_attribute_name} from credential
+     * and rename to {@code attribute_name}.
+     */
+    private List<CognitoMappedAttribute> mappedAttributes = new ArrayList<>(0);
+
+    @RequiresModule(name = "cas-server-support-aws-cognito-authentication")
+    @Getter
+    @Setter
+    @Accessors(chain = true)
+    public static class CognitoMappedAttribute implements Serializable {
```

Comment:
If you just have a key and a value, why not use an actual Map? Doesn't seem like the inner type is needed at all.

Reply:
I think inner type allows more explicit mapping configuration as given below:
```
cas.authn.cognito.mapped-attributes[0].name=custom:netid
cas.authn.cognito.mapped-attributes[0].mapped-to=netid 
```
I got this idea from pac4j saml delegated auth configuration settings.

Reply:
Indeed, but that is unnecessary here. The SAML attribute is capable of mapping more than to just a name and can do more. This is just a 1-1 relationship and is overkill here.

Reply:
Furthermore, given the changes in upstream master, an inner static class for a configuration properly most likely will produce unexpected results when collecting metadata for settings. So it's best to generally remove this and go for the simple case.

Reply:
Got it. I will go with a simple map.